x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/action/EsqlQueryRequest.java

```diff
@@ -65,17 +67,47 @@ public EsqlQueryRequest(StreamInput in) throws IOException {
     @Override
     public ActionRequestValidationException validate() {
         ActionRequestValidationException validationException = null;
+        if (Strings.hasText(esqlVersion) == false) {
+            // TODO: make this required
+            // "https://github.com/elastic/elasticsearch/issues/104890"
+            // validationException = addValidationError("[" + RequestXContent.ESQL_VERSION_FIELD + "] is required", validationException);
+        } else {
+            EsqlVersion version = EsqlVersion.parse(esqlVersion);
```

Comment:
shouldn't we be storing this version somewhere? how is it going to get passed into the actual query processing?

Reply:
For now, this only validates the version and does nothing with it.

I plan to address this in a follow-up PoC, where I want to see the version mechanism in action with a (simulated) breaking change.

We don't want to pass the raw (or parsed) version around, we want to pass behavior flags that are associated with a given version. These are going to be retrieved from the parsed query and passed to the corresponding components (esp. to the planner/optimizer).

I'm unsure where exactly these are going to be stored (or if at all). In practice, we'll need to retrieve the behavior flags from the `EsqlQueryRequest`'s `esql.version` in [EsqlSession.execute](https://github.com/elastic/elasticsearch/blob/76f142f454f092d04273fe65a4b8407d0cb0e39c/x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/session/EsqlSession.java#L122), I think, and there inform the relevant components of their expected behavior.